app/utils/metadata.py

```diff
@@ -635,6 +662,33 @@ def compile_metadata(self, uuids: list[str] = []) -> None:
                                 dependencies_by_ver["li"],
                                 self.internal_local_metadata,
                             )
+                            # Cleanup Dependencies so that it can be used as load order rules
+                            if (
+                                self.settings_controller.settings.dependency_for_sorting
+                                == "True"
+                            ):
+                                load_dependencies_by_ver = (
+                                    str(dependencies_by_ver["li"])
+                                    .lower()
+                                    .split(",")[0]
+                                    .replace("{'packageid': '", "")
+                                    .replace("[{'packageid' :", "")
+                                    .replace("displayname': '", "")
+                                    .replace("[", "")
+                                    .replace("'", "")
+                                    .replace("{", "")
+                                    .replace("}", "")
+                                    .replace("li:", "")
+                                )
+                                logger.debug(load_dependencies_by_ver)
+                                add_load_rule_to_mod(
+                                    self.internal_local_metadata[uuid],
+                                    load_dependencies_by_ver,
+                                    "loadTheseBefore",
+                                    "loadTheseAfter",
+                                    self.internal_local_metadata,
+                                    self.packageid_to_uuids,
+                                )
```

Comment:
Duplicated code. Extract to a private method

Reply:
Thanks for Suggesstions
This is the actual part of code that can be improved and i am clueless to how
thats the reason the code is duplicated

Reply:
ok now can you check